doc/centos-tutorial.txt

```diff
@@ -0,0 +1,141 @@
+tutorial will assume you have them.
+
+A. INSTALLING AND COMPILING MANA
+
+1. Installing essential packages - this section can be skipped depending on the
+existing packages on your machine.
+
+  # yum install -y git
+  # yum groupinstall 'Development Tools'
+  # yum install -y mlocate vim wget
+  # yum install -y centos-release-scl
+  # yum install -y devtoolset-8-gcc devtoolset-8-gcc-c++
+  # echo 'source scl_source enable devtoolset-8' > ~/.bashrc
+  # source ~/.bashrc
+  # yum install -y glibc-static.x86_64
+  # wget \
+      https://www.mpich.org/static/downloads/3.3.2/mpich-3.3.2.tar.gz \
+      --no-check-certificate
+  # tar xzf mpich-3.3.2.tar.gz
+  # cd mpich-3.3.2
+  # ./configure
+  # make -j install
+  # yum install -y libxml2-static zlib-static xz-devel.x86_64 yum-utils.noarch
+  # mkdir liblzma_TEMP && cd liblzma_TEMP
+  # yumdownloader --source xz-devel
+  # rpm2cpio xz-5.2.2-1.el7.src.rpm | cpio -idv
+  # tar xf xz-5.2.2.tar.gz
+  # cd xz-5.2.2
+  # patch -Np1 < ../xz-5.2.2-compat-libs.patch
+  # patch -Np1 < ../xz-5.2.2-man-page-day.patch
+  # ./configure --enable-static
+  # make
+  # cp ./src/liblzma/.libs/liblzma.a /usr/lib64/
+  # cd ~
+
+2. Installing the MANA directory and compiling MANA
+
+  $ git clone https://github.com/mpickpt/mana.git
+  $ cd mana
+  $ git checkout feature/centos
+  
+Here, we need to make some changes to the configure-mana file - these changes
+will eventually be included in a new pull request and merged into this branch.
+For now, it is included as a patch in the root directory.
+
+  $ git apply configure-mana.patch
+  $ ./configure-mana
+  $ make -j mana
+
+This should build MANA smoothly.
+
+B. LAUNCHING, CHECKPOINTING AND RESTARTING A MPI APPLICATION
+
+We'll first add the bin directory to PATH. Ensure that these commands are done
+from within the root directory of MANA.
+
+  $ echo 'export PATH=$PATH:{PATH TO MANA}/bin' >> ~/.bashrc
+  $ source ~/.bashrc
+
+By default, this should be /home/{YOUR USERNAME}/mana.
+
+1. Launching an MPI application
+
+The MANA directory comes with many test MPI applications that can be found in
+contrib/mpi-proxy-plugin/test. Depending on the application, we may require
+more than 1 copy of the process running - for example, ping_pong.mana.exe
+requires 2. To support this, change the argument after -np accordingly. For this
+tutorial, we'll use mpi_hello_world.mana.exe, which only requires 1 copy
+running.
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_launch \ 
+      contrib/mpi-proxy-split/test/mpi_hello_world.mana.exe
+
+If the application is launched properly, you should see the following printed:
+
+  Hello world from processor test, rank 0 out of 1 processors
+  Will now sleep for 500 seconds ...
+
+From here, we can either exit the program (through CTRL-C), or continue on to
+the next set of instructions to checkpoint and restart the application.
+
+2. Checkpointing and restarting a MPI application
```

Comment:
Let's split this into `2. Checkpointing an MPI application` and `3. Restarting an MPI application`, and have two seaprate sections.